tgui/packages/tgui/interfaces/Orbit.tsx

```diff
@@ -0,0 +1,366 @@
+                {name} ({section.length})
+              </Box>
+            }>
+            <ObservableMap color={color} index={index} section={section} />
+          </Section>
+        </Stack.Item>
+      )
+    );
+  });
+};
+
+/** Displays a collapsible section for ghosts, NPCs, etc. */
+const CollapsingSections = (props, context) => {
+  const { sections = [] } = props;
+
+  return sections?.map((section, index) => {
+    const { color, name } = COLLAPSING_TITLES[index];
+    return (
+      !!section.length && (
+        <Stack.Item key={index}>
+          <Collapsible
+            color={color}
+            title={name + ` (${section.length})`}
+            section={section}>
+            <ObservableMap color={color} section={section} />
+          </Collapsible>
+        </Stack.Item>
+      )
+    );
+  });
+};
+
+/** Displays a map of observable items */
+const ObservableMap = (props, context) => {
+  const { act } = useBackend<Data>(context);
+  const { color, section = [] } = props;
+  const [autoObserve, setAutoObserve] = useLocalState<boolean>(
+    context,
+    'autoObserve',
+    false
+  );
+
+  return section?.map((observable, index) => {
+    const { name, orbiters, ref } = observable;
+    const threat = getThreat(orbiters);
+    return (
+      <Button
+        color={threat || color}
+        key={index}
+        mt={0}
+        mb={0}
+        onClick={() => act('orbit', { auto_observe: autoObserve, ref: ref })}>
+        <Table>
+          <Table.Row>
+            <Table.Cell>{nameToUpper(name).slice(0, 44)}</Table.Cell>
+            {orbiters && (
+              <>
+                <Table.Cell>({orbiters.toString()}</Table.Cell>
+                <Table.Cell>
+                  <Icon
+                    mr={0}
+                    name={threat === THREAT.Large ? 'skull' : 'ghost'}
+                  />
+                  )
+                </Table.Cell>
+              </>
+            )}
+          </Table.Row>
+        </Table>
+      </Button>
+    );
+  });
+};
+
+/** Takes the amount of orbiters and returns some style options */
+const getThreat = (orbiters: number): THREAT => {
+  if (!orbiters || orbiters <= 2) {
+    return THREAT.None;
+  } else if (orbiters === 3) {
+    return THREAT.Small;
+  } else if (orbiters <= 6) {
+    return THREAT.Medium;
+  } else {
+    return THREAT.Large;
+  }
+};
+
+/**
+ * Filters both lists for the search query.
+ *
+ * Returns:
+ *  an array of Observable[]
+ */
+const getFilteredLists = (
+  lists: Observable[][],
+  searchQuery: string
+): Observable[][] => {
+  let filteredLists: Observable[][] = [];
+  lists.map((list) => {
+    const filtered = list.filter((observable) => {
+      return observable.name
+        ?.toLowerCase()
+        ?.includes(searchQuery?.toLowerCase());
+    });
+    filteredLists.push(filtered);
+  });
+  return filteredLists;
+};
+
+/** Sorts by the highest orbiter count */
+const sortByOrbiters = (a: Observable, b: Observable): number => {
```

Comment:
Use sortBy instead and just a basic key

Reply:
I ended up using sortBy elsewhere where i had guaranteed props (antagonist.antag), but this had trouble where observable.observers is sometimes undefined. 
I reckon the old fashion sort is much more concise and readable in this case

Also, toying with sortBy has clued me into how the common packages (flow, map, filter) work, and I'm super thankful for that

Reply:
nvm sortby superior